uaclient/cli.py

```diff
@@ -1068,36 +1034,16 @@ def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
             raise exceptions.UserFacingError(
                 ua_status.MESSAGE_DETACH_AUTOMATION_FAILURE
             )
-    contract_client = contract.UAContractClient(cfg)
-    try:
-        tokenResponse = contract_client.request_auto_attach_contract_token(
-            instance=instance
-        )
-    except contract.ContractAPIError as e:
-        if e.code and 400 <= e.code < 500:
-            raise exceptions.NonAutoAttachImageError(
-                ua_status.MESSAGE_UNSUPPORTED_AUTO_ATTACH
-            )
-        raise e
-    if current_iid:
-        cfg.write_cache("instance-id", current_iid)
 
-    return tokenResponse["contractToken"]
-
-
-@assert_root
-@assert_lock_file("ua auto-attach")
-def action_auto_attach(args, *, cfg):
-    disable_auto_attach = util.is_config_value_true(
-        config=cfg.cfg, path_to_value="features.disable_auto_attach"
-    )
-    if disable_auto_attach:
-        msg = "Skipping auto-attach. Config disable_auto_attach is set."
-        logging.debug(msg)
-        print(msg)
+    try:
+        actions.auto_attach(cfg, instance)
+        _post_cli_attach(cfg)
```

Comment:
We could call the `_post` function in the else block of the `try..except`
Also, you are create those helper functions just for readability purposes ? If yes, might we could run their logic here directly

Reply:
True I can move it to `else` :+1:

`_post_cli_attach` is also called from  `action_attach` below so I'd rather avoid duplicating the logic